### ISSUE TITLE
ScaledObject not created/found

### ISSUE THREAD

**Issue by blackmann** [kedacore/http-add-on#188]
The `<deploy>-external-scaler` and `<deploy>-interceptor-proxy` vanish after few seconds of creating `httpscaledobject`

## Expected Behavior
`ScaledObject` should be created so that the scaler and proxy can stay.

## Actual Behavior
`ScaledObject` does not get created. The following log is pulled from `kubectl logs keda-add-ons-http-controller-manager-<hash> keda-addons-operator`

```
2021-06-19T21:42:43.395Z	INFO	controllers.HTTPScaledObject	Creating external scaler Deployment	{"reconciler.appObjects": "addObjects", "HTTPScaledObject.name": "xkcd", "HTTPScaledObject.namespace": "default", "Deployment": {"kind":"Deployment","metadata":{"name":"xkcd-external-scaler","namespace":"default","creationTimestamp":null,"labels":{"app":"kedahttp-xkcd-external-scaler","name":"xkcd-external-scaler"}},"spec":{"replicas":1,"selector":{"matchLabels":{"app":"kedahttp-xkcd-external-scaler","name":"xkcd-external-scaler"}},"template":{"metadata":{"creationTimestamp":null,"labels":{"app":"kedahttp-xkcd-external-scaler","name":"xkcd-external-scaler"}},"spec":{"containers":[{"name":"xkcd-external-scaler","image":"ghcr.io/kedacore/http-add-on-scaler:canary","ports":[{"containerPort":8091}],"env":[{"name":"KEDA_HTTP_SCALER_PORT","value":"8091"},{"name":"KEDA_HTTP_HEALTH_PORT","value":"8092"},{"name":"KEDA_HTTP_SCALER_TARGET_ADMIN_NAMESPACE","value":"default"},{"name":"KEDA_HTTP_SCALER_TARGET_ADMIN_SERVICE","value":"xkcd-interceptor-admin"},{"name":"KEDA_HTTP_SCALER_TARGET_ADMIN_PORT","value":"8090"}],"resources":{},"livenessProbe":{"httpGet":{"path":"/livez","port":8092},"periodSeconds":1},"readinessProbe":{"httpGet":{"path":"/healthz","port":8092},"periodSeconds":1},"imagePullPolicy":"Always"}]}},"strategy":{}},"status":{}}}
2021-06-19T21:42:43.412Z	INFO	controllers.HTTPScaledObject	Creating external scaler Service	{"reconciler.appObjects": "addObjects", "HTTPScaledObject.name": "xkcd", "HTTPScaledObject.namespace": "default", "Service": {"kind":"Service","metadata":{"name":"xkcd-external-scaler","namespace":"default","creationTimestamp":null,"labels":{"app":"kedahttp-xkcd-external-scaler","name":"xkcd-external-scaler"}},"spec":{"ports":[{"name":"externalscaler","protocol":"TCP","port":8091,"targetPort":8091}],"selector":{"app":"kedahttp-xkcd-external-scaler","name":"xkcd-external-scaler"},"type":"ClusterIP"},"status":{"loadBalancer":{}}}}
2021-06-19T21:42:46.251Z	INFO	controllers.HTTPScaledObject	Updating status on HTTPScaledObject	{"HTTPScaledObject.Namespace": "default", "HTTPScaledObject.Name": "xkcd", "resource version": "77123"}
2021-06-19T21:42:46.292Z	INFO	controllers.HTTPScaledObject	Updated status on HTTPScaledObject	{"HTTPScaledObject.Namespace": "default", "HTTPScaledObject.Name": "xkcd", "resource version": "77200"}
2021-06-19T21:42:46.292Z	ERROR	controllers.HTTPScaledObject	Removing app resources	{"HTTPScaledObject.Namespace": "default", "HTTPScaledObject.Name": "xkcd", "error": "Scaler failed to start up within 2500000000"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
github.com/kedacore/http-add-on/operator/controllers.(*HTTPScaledObjectReconciler).Reconcile
	/go/src/github.com/kedahttp/http-add-on/operator/controllers/httpscaledobject_controller.go:127
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:298
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.UntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99
2021-06-19T21:42:46.444Z	INFO	controllers.HTTPScaledObject	App ScaledObject not found, moving on	{"reconciler.appObjects": "removeObjects", "HTTPScaledObject.name": "xkcd", "HTTPScaledObject.namespace": "default"}
2021-06-19T21:42:46.446Z	INFO	controllers.HTTPScaledObject	Interceptor ScaledObject not found, moving on	{"reconciler.appObjects": "removeObjects", "HTTPScaledObject.name": "xkcd", "HTTPScaledObject.namespace": "default"}
2021-06-19T21:42:46.446Z	INFO	controllers.HTTPScaledObject	Updating status on HTTPScaledObject	{"HTTPScaledObject.Namespace": "default", "HTTPScaledObject.Name": "xkcd", "resource version": "77200"}
2021-06-19T21:42:46.499Z	INFO	controllers.HTTPScaledObject	Updated status on HTTPScaledObject	{"HTTPScaledObject.Namespace": "default", "HTTPScaledObject.Name": "xkcd", "resource version": "77212"}
2021-06-19T21:42:46.500Z	ERROR	controller-runtime.manager.controller.httpscaledobject	Reconciler error	{"reconciler group": "http.keda.sh", "reconciler kind": "HTTPScaledObject", "name": "xkcd", "namespace": "default", "error": "Scaler failed to start up within 2500000000"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:302
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:253
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.3/pkg/internal/controller/controller.go:216
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:155
k8s.io/apimachinery/pkg/util/wait.BackoffUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:156
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:133
k8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:185
k8s.io/apimachinery/pkg/util/wait.UntilWithContext
	/go/pkg/mod/k8s.io/apimachinery@v0.20.4/pkg/util/wait/wait.go:99
```

## Steps to Reproduce the Problem

  1. Follow install instructions to set up the add-on
  2. Install the `xkcd` project (which comes with `httpscaledobject`)
  3. The proxy and scaler  services vanish after few seconds

## Specifications

  - **KEDA Version:** *latest*
  - **Platform & Version:** *minikube v1.21.0*
  - **Kubernetes Version:** *v1.20.7*
  - **Scaler(s):** *Please elaborate*

**Comment by blackmann** [comment]
After logs digging, it appears the wait time for scaler here [httpscaledobject_logic.go:226](https://github.com/kedacore/http-add-on/blob/929fc9707e3b2b808ff67fd0ce70503d3e1f7961/operator/controllers/httpscaledobject_logic.go#L266) is very short considering the fact that the images have to be pulled (and may exceed 5*500milliseconds = `2.5seconds`) due to network speed. And even for a fast network, there's no guarantee that the scaler failed to spawn because it may have been scheduled, (`node autoscaling`, etc).

I'd like to add that, I got it to work when I changed the `retriesCount` to 40.

**Comment by tomkerkhove** [comment]
What version are you running?

**Comment by blackmann** [comment]
All `canary`

**Comment by arschles** [comment]
thanks @blackmann - I'm gonna look into this as soon as I can 👍

**Comment by arschles** [comment]
@blackmann you mentioned in https://github.com/kedacore/http-add-on/issues/188#issuecomment-864557635 that you'd like to add something. were you referring to changing `retriesCount` to 40? if so, feel free. another change to add while you're at it, is making that hard-coded 500ms configurable via an environment variable.

what do you think?

**Comment by arschles** [comment]
>Can I work on this and submit a PR? @arschles

@blackmann absolutely, that would be wonderful. please feel free to DM me if you have any questions. I am `arschles` on the Kubernetes slack.

**Comment by arschles** [comment]
@blackmann the architecture of that portion of the codebase has changed in the (just released) `v0.2.0` release, and the operator no longer needs to wait for the scaler, so this issue wouldn't happen anymore. I'm going to close this.

If you have time and wouldn't mind, could you give `v0.2.0` a try (uninstall/reinstall according to the [install.md](https://github.com/kedacore/http-add-on/blob/main/docs/install.md)) and let me know what you find?
